File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/render/MetalBorrowedTextureDescriptor.java
package org.maplibre.nativeffi.render;

import java.util.Objects;

/** Mutable descriptor for Metal caller-owned texture render targets. */
public final class MetalBorrowedTextureDescriptor {
  private RenderTargetExtent extent = new RenderTargetExtent();
  private NativePointer texture = NativePointer.NULL;

  public RenderTargetExtent extent() {
    return extent;
  }

  public MetalBorrowedTextureDescriptor extent(RenderTargetExtent extent) {
    this.extent = Objects.requireNonNull(extent, "extent");
    return this;
  }

  public NativePointer texture() {
    return texture;
  }

  public MetalBorrowedTextureDescriptor texture(NativePointer texture) {
    this.texture = Objects.requireNonNull(texture, "texture");
    return this;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/render/MetalBorrowedTextureDescriptor.kt
package org.maplibre.nativeffi.render

/** Mutable descriptor for Metal caller-owned texture render targets. */
public class MetalBorrowedTextureDescriptor(
  extent: RenderTargetExtent = RenderTargetExtent(),
  texture: NativePointer = NativePointer.NULL,
) {
  public var extent: RenderTargetExtent = extent
    private set

  public var texture: NativePointer = texture
    private set

  public fun extent(extent: RenderTargetExtent): MetalBorrowedTextureDescriptor = apply {
    this.extent = extent
  }

  public fun texture(texture: NativePointer): MetalBorrowedTextureDescriptor = apply {
    this.texture = texture
  }
}
